src/App.tsx

```diff
@@ -1,61 +1,111 @@
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import { useState } from 'react';
+
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+import TodoWithUser from './types/TodoWithUser';
+import { TodoList } from './components/TodoList';
+
+import findUserById from './services/findUserById';
+import findMaxUserId from './services/findMaxUserId';
+import getTodosWithUsers from './services/getTodosWithUsers';
 
 export const App = () => {
+  const [title, setTitle] = useState('');
+  const [hasTitleError, setHasTitleError] = useState(false);
+
+  const [selectedUserId, setSelectedUserId] = useState(0);
+  const [hasSelectedUserError, setHasSelectedUserError] = useState(false);
+
+  const [visibleTodos, setVisibleTodos] = useState<TodoWithUser[]>(
+    getTodosWithUsers(todosFromServer, usersFromServer),
+  );
+
+  function handleTitleChange(value: string) {
+    setTitle(value);
+    setHasTitleError(false);
+  }
+
+  function handleUserChange(value: string) {
+    setSelectedUserId(parseInt(value));
+    setHasSelectedUserError(false);
+  }
+
+  function onSubmit(event: React.FormEvent<HTMLFormElement>) {
+    event.preventDefault();
+
+    setHasTitleError(!title);
+    setHasSelectedUserError(selectedUserId === 0);
+
+    if (!title || selectedUserId === 0) {
+      return;
+    }
+
+    const newTodo: TodoWithUser = {
+      user: findUserById(usersFromServer, selectedUserId),
+      id: findMaxUserId(usersFromServer) + 1,
+      title: title,
+      completed: false,
+      userId: selectedUserId,
+    };
+
+    setVisibleTodos([...visibleTodos, newTodo]);
+
+    setTitle('');
+    setSelectedUserId(0);
+  }
+
   return (
     <div className="App">
       <h1>Add todo form</h1>
 
-      <form action="/api/todos" method="POST">
+      <form
+        action="/api/todos"
+        method="POST"
+        onSubmit={event => onSubmit(event)}
+      >
         <div className="field">
-          <input type="text" data-cy="titleInput" />
-          <span className="error">Please enter a title</span>
+          <span>Title: </span>
```

Comment:
Forgot to add labels for input and select.

Reply:
Thanks for noticing! Fixed my code :)